src/components/HdModal.vue

```diff
@@ -1,128 +1,93 @@
-          class="modal__actions"
+        <footer
+          class="hd-modal__footer"
+          :class="{ 'hd-modal__footer--wide': isWide }"
         >
-          <button
-            v-for="(action, i) in actions"
-            :key="i"
-            :disabled="action.disabled"
-            :class="{
-              'btn--primary': action.type === 'primary',
-              'btn--ghost': action.type !== 'primary',
-            }"
-            class="modal__actions__action btn"
-            type="button"
-            @click="action.callback"
+          <div
+            class="hd-modal__action"
+            v-for="(button, name) in actions"
+            :key="`action-${name}`"
           >
-            {{ action.label }}
-          </button>
-        </div>
+            <slot :name="name" :value="button">
+              <hd-button
+                :modifier="button.modifier"
+                :is-in-dark-background="button.isInDarkBackground"
+                :disabled="button.disabled"
+                :icon-src="button.iconSrc"
+                @click="$emit('action', button.name)"
+              >
+                {{ button.text }}
+              </hd-button>
+            </slot>
+          </div>
+        </footer>
+
         <button
           v-if="showCloseIcon"
-          :class="`modal__close--${closeIconColor}`"
-          class="modal__close"
-          @click="onCloseClick"
+          class="hd-modal__close-button"
+          @click="$emit('close')"
         >
-          <HdIcon
+          <hd-icon
             :src="closeIcon"
-            width="100%"
+            class="hd-modal__close-icon"
           />
         </button>
       </div>
-    </section>
+    </article>
   </transition>
 </template>
 
 <script>
+import HdButton from 'homeday-blocks/src/components/buttons/HdButton.vue';
 import HdIcon from 'homeday-blocks/src/components/HdIcon.vue';
 import { close as closeIcon } from 'homeday-assets';
+import { getMessages } from 'homeday-blocks/src/lang';
 
 export default {
   name: 'HdModal',
   components: {
+    HdButton,
     HdIcon,
   },
   props: {
-    closeIconColor: {
-      type: String,
-      default: 'dark',
-      validator: (closeIconColor) => ['dark', 'light'].includes(closeIconColor),
-    },
-    showCloseIcon: {
-      type: Boolean,
-      default: true,
-    },
-    externalCloseIcon: {
-      type: Boolean,
-      default: false,
-    },
-    overlayColor: {
-      type: String,
-      default: 'dark',
-      validator: (overlayColor) => ['dark', 'light'].includes(overlayColor),
-    },
-    width: {
-      type: Number,
-      default: 0,
-      validator: (maxWidth) => maxWidth >= 0,
-    },
-    spacingSize: {
+    iconSrc: {
       type: String,
-      default: 'large',
-      validator: (spacingSize) => ['large', 'normal'].includes(spacingSize),
+      default: '',
     },
-    noSpacing: {
+    isWide: {
       type: Boolean,
       default: false,
     },
-    noExternalSpacing: {
+    showCloseIcon: {
```

Comment:
Also, don't forget Ilyas's previous suggestion about renaming this prop.

> And let's use the same naming convention for showCloseIcon —which might be mistaken for a function— (e.g. isClosable or isCloseButtonVisible).

Reply:
https://github.com/homeday-de/homeday-blocks/pull/732/commits/9fa03222c4b4165db9c1b831f9dd2748d4e89a83 @ilyasmez - thanks for reminding me bro <3